### ISSUE TITLE
Intermittent failures on GraphQL plugin with TypeError: field.span._getTime is not a function

### ISSUE THREAD

**Issue by shayonj** [DataDog/dd-trace-js#1163]
**Describe the bug**
Intermittent failures on GraphQL plugin with `TypeError: field.span._getTime is not a function`

```
[3] (node:23229) UnhandledPromiseRejectionWarning: TypeError: field.span._getTime is not a function
[3]     at updateField (/Users/shayon/src/loom/node_modules/dd-trace/packages/datadog-plugin-graphql/src/index.js:349:33)
[3]     at /Users/shayon/src/loom/node_modules/dd-trace/packages/datadog-plugin-graphql/src/index.js:158:62
[3]     at /Users/shayon/src/loom/node_modules/dd-trace/packages/datadog-plugin-graphql/src/index.js:176:16
[3]     at runNextTicks (internal/process/task_queues.js:58:5)
[3]     at processImmediate (internal/timers.js:434:9)
[3]     at process.topLevelDomainCallback (domain.js:138:15)
[3]     at process.callbackTrampoline (internal/async_hooks.js:127:14)
[3] (node:23229) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
[3] (node:23229) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

On `v0.29.0` we saw a high rate of the above errors when rolled out. We were on a much older version. I think its stemming after this https://github.com/DataDog/dd-trace-js/pull/1139/ and might have to do with `NoopSpan` being returned from here 
https://github.com/DataDog/dd-trace-js/blob/c5c5f198db46e38c83f660f720315ae5dab32f63/packages/dd-trace/src/opentracing/tracer.js#L84

We have sampling turned on. 

**Environment**
Locally and Production
* **Operation system:**
Mac OSX (also in production on Amazon Linux 2)

* **Node version:**
14.15
* **Tracer version:**
N/A
* **Agent version:**
N/A

**Comment by shayonj** [comment]
One fix would be to keep `NoopSpan` and `DatadogSpan` similar? IOW, porting the `_getTime` function to `NoopSpan` class? Or accounting for two different spans from within `updateField` in `packages/datadog-plugin-graphql/src/index.js`

**Comment by rochdev** [comment]
I'm implementing a fix right now directly in the plugin. Ideally it shouldn't rely on a private property of the span in the first place, but it's not trivial since the wall time is stored on the trace.

**Comment by shayonj** [comment]
Wonderful! Fixing it in the plugging makes sense, happy to cut a PR, lmk. Otherwise, really appreciate for the quick workaround. Thanks so much 🙏

**Comment by rochdev** [comment]
Fixed in 0.29.1
